src/parser/ast.c

```diff
@@ -121,8 +123,34 @@ AST_Validation _Validate_RETURN_Clause(const AST_Query *ast, char **reason) {
   TrieMap *return_aliases = NewTrieMap();
   ReturnClause_ReferredNodes(ast->returnNode, return_aliases);
   
-  AST_Validation res = _Validate_Aliases_In_Match_Clause( return_aliases, ast->matchNode, reason);
+  AST_Validation res = _Validate_Aliases_In_Match_Clause(return_aliases, ast->matchNode, reason);
   TrieMap_Free(return_aliases, TrieMap_NOP_CB);
+  if(res != AST_VALID) return res;
+
+  // Make sure all referenced function do exists.
+  TrieMap *ref_functions = NewTrieMap();
+  ReturnClause_ReferredFunctions(ast->returnNode, ref_functions);
+
+  void *value;
+  tm_len_t len;
+  char *refFuncName;
+  char funcName[1024];
+  TrieMapIterator *it = TrieMap_Iterate(ref_functions, "", 0);
+
+  // TODO: return RAX.
+  while(TrieMapIterator_Next(it, &refFuncName, &len, &value)) {
+    if(len >= 1023) len = 1023;
```

Comment:
If `len >= 1023`, I think we can just call the referred function invalid, right?